### DISCUSSION TITLE
Hookrouter vs raviger || Downgrade to React 17?

### DISCUSSION THREAD

**Discussion by dustinboettcher** [kinngh/shopify-node-express-mongodb-app#9] [General]
First of all, excellent template. 

I just wonder why you are working with force install.

Hookrouter seems to be more or less shut down. You could replace it with raviger (https://www.npmjs.com/package/raviger) without having to make big changes.

Of course React 18 is cutting edge, with 17 the package runs through without errors.

Best Dustin

**Comment by kinngh** [comment]
Thank you!

So I've been using `hookrouter` for a while and it serves it's function well, and the idea is to be on the cutting edge version(s) of every package we use so we're leveraging the newest standards and don't worry about thrown into unsupported territory overnight and scramble to update and make changes. I'll take a look at `raviger`, and I've also been exploring other alternatives (and possibly going native with routing) and going to make a call on this next week! 


The repo is still WIP so I have that flexibility to make changes, but we'll see next week if we setup a new solution or fork `hookrouter` for a new routing solution 👀

**Reply by dustinboettcher** [reply]
> ...(and possibly going native with routing)

I was thinking the same thing while testing. I just read that you use the hookrouter in other projects as well, that's where I came up with raviger. That would be only a small change.

Will definitely keep an eye on the repo. Thumbs up =)

**Reply by kinngh** [reply]
Oh I am actively looking into making the repo more stable and mature, so if going native with routing makes sense, it's easier to make a few hooks / functions to get it up and running that reduces the amount of boilerplate code that's going it, it'll be amazing!

I really appreciate the feedback, Dustin!

**Reply by kinngh** [reply]
I've switched over to `raviger` for now, since it just fixes a lot of bugs straight up! I'll be looking more into routing solutions and make a change, if required, next week!

**Reply by dustinboettcher** [reply]
> ...since it just fixes a lot of bugs straight up!

That is great! I am glad that it works.
